### ISSUE TITLE
landmark tools

### ISSUE THREAD

**Issue by robustfengbin** [clcarwin/sphereface_pytorch#4]
did you have the landmark tools which get landmark from a new photo

**Comment by clcarwin** [comment]
[mtcnn](https://github.com/pangyupo/mxnet_mtcnn_face_detection) has been used to get face landmark.

**Comment by xiangjun0103** [comment]
I have used mtcnn，but don't get the same landmark points value of photo from CASIA-maxpy-clean.zip with casia_landmark.txt of the git project

**Comment by radonys** [comment]
I would like to know the order of landmarks written in the casia_landmark.txt so that I can write in the same way for my own dataset and use your defined functions.

**Comment by mengjunleng** [comment]
I have tried the mtcnn. However, the landmark location detected are slightly different from what you provided under the data folder for both Webface and LFW. Do you use only other processing to refine the landmark location?

**Comment by Rigel-1994** [comment]
@radonys, I have the same problem with you. Did you figure out the landmark order?

**Comment by Mukosame** [comment]
Yes, like for the first face _Aaron_Eckhart_0001.jpg_ the landmarks I generate with [MTCNN](https://github.com/TropComplique/mtcnn-pytorch) is 

`[[106.54915  147.81348  124.34372  105.037575 139.50761  108.094376
  112.37028  142.72244  158.04742  161.84981 ]]`

While the _lfw_landmark.txt_ file says the landmark is:

`Aaron_Eckhart/Aaron_Eckhart_0001.jpg	107	107	147	111	124	143	103	157	139	161`

Ignoring the float to int method, there is still slight difference between these numbers. But I think the order could be:

`txt[0] = mtcnn[0]`
`txt[1] = mtcnn[5]`
`txt[2] = mtcnn[1]`
`txt[3] = mtcnn[6]`
`txt[4] = mtcnn[2]`
`txt[5] = mtcnn[7]`
`txt[6] = mtcnn[3]`
`txt[7] = mtcnn[8]`
`txt[8] = mtcnn[4]`
`txt[9] = mtcnn[9]`

**Comment by zorrocai** [comment]
@Mukosame I met the same problem as you. The landmarks generated by MTCNN are different from the landmarks provided. Have you solved the problem?

**Comment by DecentMakeover** [comment]
Hi @Mukosame , @radonys,  @clcarwin ,  @mengjunleng 
I see that the landmark points are compared with source points, any body has any ideas as how to calculate source points for my own dataset?

Thanks in advance.

**Comment by noamgat** [comment]
@Mukosame 's solution looks to be working (its a transpose of a 2x5 matrix). However, when I try to replace the pre-baked landmark data with any python MTCNN library, I get accuracy of 96% (as opposed to 99% when using the data file). Has anyone been able to reproduce the 99% accuracy with calculated landmarks?
